@here/olp-sdk-core/lib/utils/multipartupload-internal/NodeFileData.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { BlobData, fs } from "@here/olp-sdk-core";
+
+/**
+ * @internal
+ * Implementation of reading bytes from file by filepath
+ * in Node.js.
+ */
+export class NodeFileData implements BlobData {
+    private fileHandle?: fs.promises.FileHandle;
+
+    constructor(private readonly filePath: string) {}
+
+    async readBytes(offset: number, count: number): Promise<ArrayBuffer> {
+        if (!this.fileHandle) {
+            this.fileHandle = await this.openFile();
+        }
+        const fileSize = await this.size();
+        const bufferLength =
+            offset + count > fileSize ? fileSize - offset : count;
+
+        const buffer = Buffer.alloc(bufferLength);
+        const chunk = await this.fileHandle.read(
+            buffer,
+            0,
+            bufferLength,
+            offset
+        );
+        return chunk.buffer;
+    }
+
+    async size(): Promise<number> {
+        if (!this.fileHandle) {
+            this.fileHandle = await this.openFile();
+        }
+        const stat = await this.fileHandle.stat();
+        return stat.size;
+    }
+
+    private async openFile() {
+        return fs.promises.open(this.filePath, "r");

```

Comment:
I don't see where you close the file. Does the fileHandle close it automatically on destructor or why dont you need to call fileHandle.close().

According to the documentation:
If a <FileHandle> is not closed using the filehandle.close() method, it will try to automatically close the file descriptor and emit a process warning, helping to prevent memory leaks. Please do not rely on this behavior because it can be unreliable and the file may not be closed. Instead, always explicitly close <FileHandle>s. Node.js may change this behavior in the future.

See https://nodejs.org/api/fs.html#fs_class_filehandle

Reply:
Added `finally` method, will call it after all file was reading. We don't have possibility to call it on the destruction of the `BlobData`